src/orch/manager.c

```diff
@@ -0,0 +1,232 @@
+}
+
+bool manager_set_port(Manager *manager, const char *port_s) {
+        uint16_t port = 0;
+
+        if (!parse_port(port_s, &port)) {
+                fprintf(stderr, "Invalid port format '%s'\n", port_s);
+                return false;
+        }
+        manager->port = port;
+        return true;
+}
+
+bool manager_parse_config(Manager *manager, const char *configfile) {
+        _cleanup_hashmap_ struct hashmap *ini_hashmap = NULL;
+        struct hashmap *manager_hashmap = NULL;
+        const char *port = NULL;
+
+        ini_hashmap = parsing_ini_file(configfile);
+        if (ini_hashmap == NULL) {
+                return false;
+        }
+
+        manager_hashmap = hashmap_get(ini_hashmap, "Manager");
+        if (manager_hashmap == NULL) {
+                return true;
+        }
+
+        port = hashmap_get(manager_hashmap, "Port");
+        if (port) {
+                if (!manager_set_port(manager, port)) {
+                        return false;
+                }
+        }
+
+        return true;
+}
+
+static int manager_accept_node_connection(
+                UNUSED sd_event_source *source, int fd, UNUSED uint32_t revents, void *userdata) {
+        Manager *manager = userdata;
+        ManagedNode *node = NULL;
+        _cleanup_fd_ int nfd = accept_tcp_connection_request(fd);
+
+        if (nfd < 0) {
+                return -1;
+        }
+
+        _cleanup_sd_bus_ sd_bus *dbus_server = peer_bus_open_server(
+                        manager->event, "managed-node", HIRTE_DBUS_NAME, steal_fd(&nfd));
+        if (dbus_server == NULL) {
+                return -1;
+        }
+
+        node = managed_node_new(manager, dbus_server);
+        if (dbus_server == NULL) {
+                return -1;
+        }
+
+        LIST_APPEND(nodes, manager->nodes, node);
+
+        return 0;
+}
+
+static bool manager_setup_node_connection_handler(Manager *manager) {
+        int r = 0;
+        _cleanup_sd_event_source_ sd_event_source *event_source = NULL;
+
+        _cleanup_fd_ int accept_fd = create_tcp_socket(manager->port);
+        if (accept_fd < 0) {
+                return false;
+        }
+
+        r = sd_event_add_io(
+                        manager->event, &event_source, accept_fd, EPOLLIN, manager_accept_node_connection, manager);
+        if (r < 0) {
+                fprintf(stderr, "Failed to add io event: %s\n", strerror(-r));
+                return false;
+        }
+        r = sd_event_source_set_io_fd_own(event_source, true);
+        if (r < 0) {
+                fprintf(stderr, "Failed to set io fd own: %s\n", strerror(-r));
+                return false;
+        }
+
+        // sd_event_set_io_fd_own takes care of closing accept_fd
+        steal_fd(&accept_fd);
+
+        (void) sd_event_source_set_description(event_source, "node-accept-socket");
+
+        manager->node_connection_source = steal_pointer(&event_source);
+
+        return true;
+}
+
+bool manager_start(Manager *manager) {
```

Comment:
There should be a NULL pointer check here:
```
assert(manager);
```

Reply:
I don't get why we are sprinkling these checks all over the place? Its like having a c++ code where every method asserts that this != null? Just makes things more verbose. This function is gonna be called once at startup of the orchestrator and never more, and it clearly is not-null in that case.